common/djangoapps/student/helpers.py

```diff
@@ -122,7 +124,10 @@ def check_verify_status_by_course(user, course_enrollment_pairs, all_course_mode
         # verification status.
         if verified_mode is not None and enrollment.mode in CourseMode.VERIFIED_MODES:
             deadline = verified_mode.expiration_datetime
+
             relevant_verification = SoftwareSecurePhotoVerification.verification_for_datetime(deadline, verifications)
+            if relevant_verification is not None:
+                most_recent_verifications.append(relevant_verification.expiration_datetime)
```

Comment:
@wedaly you mean some thing like this.  Compare the value and store it in a variable.
if most_recent_verification is None:
        most_recent_verification = relevant_verification.expiration_datetime
elif relevant_verification.expiration_datetime > most_recent_verification:
     most_recent_verification = relevant_verification.expiration_datetime

Earlied i used above technique but list method was more clean and simple. But due to space efficiency i will remove that.